vendor/bat-native-ledger/src/bat/ledger/internal/contribution/contribution.h

```diff
@@ -294,6 +296,7 @@ class Contribution {
   uint32_t last_reconcile_timer_id_;
   std::map<std::string, uint32_t> retry_timers_;
   uint32_t queue_timer_id_;
+  bool queue_in_progress_ = false;
```

Comment:
Added this bool to know that we are not triggering contribution for the same queue item

Reply:
`queue_in_progress_` state is prone to bugs. This could be resolved by removing the item from the queue when triggering an item and then re-adding to the end of the queue if a failure occurs

Reply:
we can't do that as we could loose data. For example let's say that you delete it and user closes/crash the browser

Reply:
Each queue entry, therefore, needs a state, i.e. `ready` and `processing`, this way you can find items which are waiting and ready to be processed and delete `processed` items using an id

Reply:
not sure what would be a benefit of this? We run this process in one thread, and it's only triggered here. I understand where you are coming from, but maybe it's too complex what you are saying

Reply:
we only handle one item in queue at the time from the db

Reply:
And this is guaranteed on iOS too?

Reply:
yup one item at the time on iOS as well